lib/src/fixed_datetime_formatter.dart

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection';
+
+/// A class for parsing and formatting dates for a fixed format string. For
+/// example, calling
+/// `FixedDateTimeCodec('YYYYMMDDhhmmss').decodeToLocal('19960425050322')` has
+/// the same result as calling `DateTime(1996, 4, 25, 5, 3, 22)`.
+///
+/// The allowed characters are
+/// * `Y`	a digit used in the time scale component “calendar year”
+/// * `M`	a digit used in the time scale component “calendar month”
+/// * `D`	a digit used in the time scale component “calendar day”
+/// * `E`	a digit used in the time scale component “decade”
+/// * `C`	a digit used in the time scale component “century”
+/// * `h`	a digit used in the time scale component “clock hour”
+/// * `m`	a digit used in the time scale component “clock minute”
+/// * `s`	a digit used in the time scale component “clock second”
+/// as specified in the ISO 8601 standard.
+///
+/// Non-allowed characters in the format [pattern] are ignored when decoding a
+/// string, in this case `YYYY kiwi MM` is the same format string as
+/// `YYYY------MM`. When encoding a datetime, the non-format characters are in
+/// the output verbatim.
+///
+/// Note: this class differs from [DateFormat] in that here, the characters are
+/// treated literally, i.e. the format string `YYY` matching `996` would result in
+/// the same as calling `DateTime(996)`. [DateFormat] on the other hand uses the
+/// specification in https://www.unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table,
+/// the format string (or "skeleton") `YYY` specifies only the padding, so
+/// `1996` would be a valid match. This limits is use to format strings
+/// containing delimiters, as the parser would not know how many digits to take
+/// otherwise.
+class FixedDateTimeFormatter {
+  static const _validFormatCharacters = 'YMDEChms';
+
+  final String pattern;
+  // ignore: prefer_collection_literals
+  final _occurences = LinkedHashMap<String, _Range>();
```

Comment:
The lint is valid here, the map literal will create a LinkedHashMap. It shouldn't matter whether the field type here is LinkedHashMap instead of Map.

There is also a spelling error -> occurrences.

```suggestion
  final _occurrences = <String, _Range>{};
```

Reply:
I thought it does, as I want to preserve insertion order, the lint confused me here. Does the normal `Map` also preserve it?

Reply:
`LinkedHashMap<String, _Range>()` and `<String, _Range>{}` have identical runtime semantics. The only difference is that `{}` has a static type of `Map`. The instance you get from the collection literal will be a `LinkedHashMap`, and it will preserve insert order.